src/pages/workspace/categories/CreateCategoryPage.tsx

```diff
@@ -0,0 +1,107 @@
+};
+
+type CreateCategoryPageProps = WorkspaceCreateCategoryPageOnyxProps & StackScreenProps<CentralPaneNavigatorParamList, typeof SCREENS.WORKSPACE.CATEGORY_CREATE>;
+
+function CreateCategoryPage({route, policyCategories}: CreateCategoryPageProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+
+    const validate = useCallback(
+        (values: FormOnyxValues<typeof ONYXKEYS.FORMS.WORKSPACE_CATEGORY_CREATE_FORM>) => {
+            const errors: FormInputErrors<typeof ONYXKEYS.FORMS.WORKSPACE_CATEGORY_CREATE_FORM> = {};
+            const categoryName = values.categoryName.trim();
+
+            if (!ValidationUtils.isRequiredFulfilled(categoryName)) {
+                errors.categoryName = 'workspace.categories.categoryRequiredError';
+            } else if (policyCategories?.[categoryName]) {
+                errors.categoryName = 'workspace.categories.existingCategoryError';
+            } else if (categoryName === CONST.INVALID_CATEGORY_NAME) {
+                errors.categoryName = 'workspace.categories.invalidCategoryName';
+            } else if ([...categoryName].length > CONST.CATEGORY_NAME_LIMIT) {
+                // Uses the spread syntax to count the number of Unicode code points instead of the number of UTF-16
+                // code units.
+                ErrorUtils.addErrorMessage(errors, 'categoryName', ['common.error.characterLimitExceedCounter', {length: [...categoryName].length, limit: CONST.TITLE_CHARACTER_LIMIT}]);
+            }
+
+            return errors;
+        },
+        [policyCategories],
+    );
+
+    const createCategory = (values: FormOnyxValues<typeof ONYXKEYS.FORMS.WORKSPACE_CATEGORY_CREATE_FORM>) => {
+        Policy.createPolicyCategory(route.params.policyID, values.categoryName.trim());
+        Keyboard.dismiss();
+        Navigation.goBack();
+    };
+
+    return (
+        <AdminPolicyAccessOrNotFoundWrapper policyID={route.params.policyID}>
+            <PaidPolicyAccessOrNotFoundWrapper policyID={route.params.policyID}>
+                <ScreenWrapper
+                    includeSafeAreaPaddingBottom={false}
+                    style={[styles.defaultModalContainer]}
+                    testID={CreateCategoryPage.displayName}
+                >
+                    <HeaderWithBackButton
+                        title={translate('workspace.categories.addCategory')}
+                        onBackButtonPress={Navigation.goBack}
+                    />
+                    <FormProvider
+                        formID={ONYXKEYS.FORMS.WORKSPACE_CATEGORY_CREATE_FORM}
+                        onSubmit={createCategory}
+                        submitButtonText={translate('common.save')}
+                        validate={validate}
+                        style={[styles.mh5, styles.flex1]}
+                    >
+                        <InputWrapper
+                            InputComponent={TextInput}
+                            maxLength={100}
```

Comment:
Here too we set the max limit as 256 based on CONST, but hardcoded as 100 here.

Reply:
Updated